.github/workflows/invalidatePRs.sh

```diff
@@ -0,0 +1,131 @@
+
+for pr in $(echo "${prs}" | jq -r '.[] | @base64'); do
+  log "$(_jq ${pr} '.title') (#$(_jq ${pr} '.number'))"
+
+  if [[ "open" != $(_jq ${pr} '.state') ]] || [[ "true" == $(_jq ${pr} '.draft') ]]; then
+    log "  Skipping due to state=$(_jq ${pr} '.state') and draft=$(_jq ${pr} '.draft')."
+    continue
+  fi
+
+  ref=$(_jq ${pr} '.base.ref')
+  sha=$(_jq ${pr} '.base.sha')
+
+  shaToCompare=${listSha[${ref}]}
+
+  # Keep the list of branch head sha
+  if [[ $shaToCompare == "" ]]
+  then
+    refCommit=$(curl -s \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: token $oauth" \
+              https://api.github.com/repos/imodeljs/imodeljs/commits/${ref})
+    refHeadCommit=$(echo ${refCommit} | jq -r '. | @base64')
+    refHeadSha=$(_jq  ${refHeadCommit} '.sha')
+    listSha[${ref}]=${refHeadSha}
+    shaToCompare=$refHeadSha
+  fi
+
+  # Compare base sha with the target branch head sha
+  log "  Head sha at ${ref} branch is ${shaToCompare} and base sha is ${sha}"
+  if [[ $shaToCompare == $sha ]]
+    then
+    log "  Skipping since it is up-to-date with the target branch, ${ref}."
+    continue
+  fi
+
+  log "  Last updated at $(_jq ${pr} '.updated_at')."
+
+  ## Check if the PR is even 3 hours old. If so, no reason to check statuses.
+  lastUpdatedTime=$(_jq ${pr} '.updated_at')
+  currentTime=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+  tooOld=$(node -e "const lutPlus3=new Date('$lastUpdatedTime'); lutPlus3.setHours(lutPlus3.getHours()+3); console.log((new Date('$currentTime') - lutPlus3) > 0)")
+  if [[ $tooOld == false ]]; then
+    log "  Skipping since it has been updated within the past 3 hours."
+    continue
+  fi
+
+  # Get all statuses of the PR
+  #   e.g. https://api.github.com/repos/imodeljs/imodeljs/statuses/c29a168b6a201052098ea6743041868bfbadaa4f
+  statuses=$(curl -s \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: token $oauth" \
+              $(_jq ${pr} '.statuses_url'))
+
+  for status in $(echo "${statuses}" | jq -r '.[] | @base64'); do
+    if [[ "success" != $(_jq  ${status} '.state') ]] || [[ "license/cla" == $(_jq  ${status} '.context') ]]; then
+      log "  Skipping $(_jq  ${status} '.context') with state $(_jq  ${status} '.state')."
+      continue
+    fi
+
+    prUpdatedTime=$(_jq  ${status} '.updated_at')
+    tooOld=$(node -e "const lutPlus3=new Date('$prUpdatedTime'); lutPlus3.setHours(lutPlus3.getHours()+3); console.log((new Date('$currentTime') - lutPlus3) > 0)")
+    if [[ $tooOld == false ]]; then
+      log "  Skipping $(_jq  ${status} '.context') it was updated within the past 3 hours."
+      continue
+    fi
+
+    echo "  The PR $(_jq ${pr} '.title') has build $(_jq  ${status} '.context') that is older than 3 hours. Invalidating..."
+
+    updateUrl=https://api.github.com/repos/imodeljs/imodeljs/statuses/$(_jq ${pr} '.head.sha')
+    target_url=$(_jq  ${status} '.target_url')
+
+    # curl \
```

Comment:
Need to uncomment this before we merge.

Reply:
@SaemJeon you should be good to uncomment this. I'll make sure the token is setup correctly.